src/main/java/io/reactivex/internal/schedulers/SchedulerPoolFactory.java

```diff
@@ -57,22 +55,25 @@ private SchedulerPoolFactory() {
      * Starts the purge thread if not already started.
      */
     public static void start() {
-        if (!PURGE_ENABLED) {
-            return;
-        }
-        for (;;) {
-            ScheduledExecutorService curr = PURGE_THREAD.get();
-            if (curr != null && !curr.isShutdown()) {
-                return;
-            }
-            ScheduledExecutorService next = Executors.newScheduledThreadPool(1, new RxThreadFactory("RxSchedulerPurge"));
-            if (PURGE_THREAD.compareAndSet(curr, next)) {
+        tryStart(PURGE_ENABLED);
```

Comment:
Factored out into a method which takes this parameter to be able to unit test the setup code. In normal code, this `start` call will usually happen once (class init).